x/concentrated-liquidity/lp.go

```diff
@@ -250,6 +250,57 @@ func (k Keeper) withdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 	return actualAmount0.Neg(), actualAmount1.Neg(), nil
 }
 
+// nolint: unused
+// addToPosition attempts to add liquidityAmount to a position with the given pool id in the given tick range.
+// Returns error if
+// - Withdrawing full position fails
+// - Creating new position with added liquidity fails
+// - Position is superfluid staked
+// TODO: handle adding to SFS positions
+func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId uint64, amount0Added, amount1Added sdk.Int) (uint64, sdk.Int, sdk.Int, error) {
```

Comment:
I don't see position ownership checked anywhere. I know that it is checked in `withdrawPosition` but I think it would be great to have it here as well to avoid depending on the internal logic downstream for such an important security check.

Also, please add a test where it fails for non-owner trying to add to a position

Reply:
Nice catch, added in